tools/packaging/common/envoy_bootstrap_v2.json

```diff
@@ -405,6 +405,7 @@
             "keepalive_time": 300
           }
         },
+        "max_requests_per_connection": 1,
```

Comment:
Sorry, I did not fully understand how this will help. Are you saying when pilot's gRPC server disconnects this Envoy - It is reconnecting on the same connection and that is why max age is not working? We do not have this and we see it is disconnected at +/- 30 mins regularly.

Reply:
I do not know the exact problem yet, but  please note, this https://github.com/envoyproxy/envoy/pull/9668 has changed the behaviour. IIUC, with that change, there will be multiple connections to same host and even if max requests per connection is hit, it might use another connection to the same host from pool. Read the circuit breaking doc updates in that PR once.

Reply:
see the linked comment in the pr description. As far as I can tell this only works today if you are connecting to pilot over plain text (port 15010). everything else is broken

Reply:
Ok. AFAICT, this should not depend on whether you are on plain text or tls. Probably the change you made to convert to gRPC should take care of that and this is not needed?

Reply:
there's 3 cases.

1. direct to pilot, using grpc stack - works
2. with pilot Sidecar - does not work. this happens when using tls on 1.4-
3 direct to pilot, using net/http stack - does not work. this happens when 1.4 proxies connect to 1.5 pilot over tls only, but no other cases. technically we could make that case use the grpc stack as well, we just have not yet

so it's not necessarily about tls or not, that just determines other factors which impact this

Reply:
case 3 is broken for all standard 1.5 uses cases without this pr as well, as otherwise the net/http stack is always used

Reply:
the max req per Conn is needed for the case with pilot Sidecar. maybe we don't care about that since we removed the Sidecar by default in 1.5

Reply:
Ok. Thanks for the details. We use 2nd case and use TLS with 1.5 master and it seems proxies are getting disconnected properly. But if you have tested and this flag does not have any side affects, we can leave it as is.